### ISSUE TITLE
BUG: Order of new clippings is lost when closing "Clippings Manager"

### ISSUE THREAD

**Issue by UrsBeeli** [aecreations/clippings#183]
My clippings are organised in the following structure:

```
+-+ Folder 1
| +-- Clipping 1.1
| +-- Clipping 1.2
| +-- Clipping 1.3
| |      .....
| +-- Clipping 1.N
|
+-+ Folder 2
| +-- Clipping 2.1
| +-- Clipping 2.2
| +-- Clipping 2.3
| |      .....
| +-- Clipping 2.N
|
+-- Clipping 3
+-- Clipping 4
|      ....
+-- Clipping N
```

When I now try to insert a new clipping within one of the two folders, it gets added at the very end (after 1.N or 2.N). If I then move it up to where it needs to be (say between 1.2 and 1.3) and close the "Clippings Manager" dialog, this change in position is lost.

This is reflected both when trying to insert a clipping in the popup-menu-structure and when re-opening the "Clippings Manager" window. In both cases, the newly added clipping is back at its original insertion point (after 1.N) instead of where it should be (between 1.2 and 1.3).

The only way I have found to solve this, is to export all clippings, then manually move the new clipping in the exported JSON file and replacing all clippings with an import from the modified JSON file.

**Comment by pyroexe** [comment]
Hi there, 

I've got the same problem. 

So my previous clippings repository (when it wasn't a web extension), it worked fine. 
But now that I have migrated to the new version, I experience this problem. 

I hope you can fix it as for me it works to classify the folders and easily locate what is needed as it's a large repository. 

Cheers!

**Comment by aecreations** [comment]
The ability to reorder/position clippings and folders will be considered for a future release.

**Comment by UrsBeeli** [comment]
@aecreations the ability to reorder/position items in the tree already exist today when working with clippings in the "Clippings Manager" dialog. The only problem is, that changes are not persistent.

**Comment by pyroexe** [comment]
I think that UrsBeeli is right, 

This is not an enhancement as the ability existed in previous versions. This is a **bug**. 

Cheers

**Comment by aecreations** [comment]
Clippings is open source, so developers are welcome to contribute enhancements and bug fixes. I would be happy to review any code you may have that brings back the ability to persist the position of clippings and folders in Clippings Manager.

**Comment by pyroexe** [comment]
Thanks for your reply, 

Further information about this. I've got sub-folders and the same idea (as the root structure) inside of them. 

They work fine in sub-folders when I've imported them from the previous version. 

However if I want to do the same in sub-folders for new clippings, it's not possible and the same behaviour is present as the root structure. 

Cheers

**Comment by aecreations** [comment]
This issue has since been fixed in version 6.1
